### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by rcoy-v**
## Changes

- Add initial integration tests around projects and occurrences. Not all existing paths are covered, this is just a starting point.
- Run integration tests in CI.
- Fix potential panic in parsing filter expressions.
- Force Elasticsearch to refresh the respective index when changes happen, so they are immediately reflected. For example, now when a project is created it could be immediately retrieved through the API. Before this could potentially fail if the requests happened fast enough, as the default refresh interval is 1 second.
- Refactor `filtering` package unit tests to Ginkgo/Gomega, to follow rest of unit tests.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,12 +6,12 @@ require (
 	github.com/Jeffail/gabs/v2 v2.6.0
 	github.com/brianvoe/gofakeit/v5 v5.10.1
 	github.com/elastic/go-elasticsearch/v7 v7.5.1-0.20201104130636-152864b47d96
-	github.com/gogo/protobuf v1.2.1
 	github.com/golang/mock v1.4.4
 	github.com/golang/protobuf v1.4.2
 	github.com/google/cel-go v0.6.0
 	github.com/google/go-cmp v0.5.2 // indirect
 	github.com/google/uuid v1.1.2
+	github.com/grafeas/client-go v0.0.0-20190625211527-39fa98b49d38
```
Comment (mrparkers) [inline go.mod:14]: probably don't need this anymore?

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -11,10 +11,22 @@ jobs:
...
+  integration:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Start services
+        run: docker-compose up -d --build elasticsearch server
+      - name: Run integration tests
+        run: make integration
+      - name: Teardown services
+        if: ${{ always() }}
```
Comment (mrparkers) [inline .github/workflows/test.yaml:31]: what's the reason for this?

File: test/v1beta1/main_test.go [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+			}
+
+			names := []string{
+				"projects/foo",
+				"projects/bar",
+				"projects/foo-bar-123",
+			}
+
+			for _, n := range names {
+				_, err = s.pc.CreateProject(s.ctx, &project_go_proto.CreateProjectRequest{Project: &project_go_proto.Project{Name: n}})
```
Comment (mrparkers) [inline test/v1beta1/main_test.go:86]: nitpick: I think this would look better on multiple lines

File: test/v1beta1/main_test.go [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+						filter:   "name==\"projects/foo\"",
+						expected: []string{"projects/foo"},
+					},
+					{
+						name:     "or exact name match",
+						filter:   "name==\"projects/foo\"||name==\"projects/bar\"",
+						expected: []string{"projects/foo", "projects/bar"},
+					},
+				} {
+					tc := tc
```
Comment (mrparkers) [inline test/v1beta1/main_test.go:103]: why is this needed?

File: .github/workflows/test.yaml (hunk 2 above)
Comment (rcoy-v) [reply .github/workflows/test.yaml:31]: For the overall teardown, I'm not familiar enough yet with github actions if it is safe to just leave docker-compose running. There is a general cleanup task the runs with the output of killing processes, but I don't know how projects/workers/etc. are reused.

The `if: ${{ always()}}` is so the the teardown step will run whether or not any other previous steps fail.

File: test/v1beta1/main_test.go (hunk 4 above)
Comment (rcoy-v) [reply test/v1beta1/main_test.go:103]: The tests in this table are run in parallel. If I don't shadow `tc`, or wrap each test in its own `func` call, they will not execute with the correct testcase.

File: test/v1beta1/main_test.go (hunk 3 above)
Comment (rcoy-v) [reply test/v1beta1/main_test.go:86]: will clean up

**Review by mrparkers — APPROVED** [review]
